src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -97,6 +111,21 @@ protected override Assembly Load(AssemblyName assemblyName)
                     return assembly;
                 }
 
+                foreach (var dependencyPath in _dependencyPaths)
+                {
+                    foreach (var extension in _extensions)
+                    {
+                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        if (IsAssemblyAlreadyLoaded(candidatePath) ||
+                            !FileMatchesAssemblyName(candidatePath, assemblyName.Name))
```

Comment:
We don't actually need `FileMatchesAssemblyName` since we're constructing the candidate path from the assembly name in the first place. We do need to test if the file exists however so we only load the one that matches (if we find one).

It only worked in my initial spike because of a fluke :) (the `.dll` was picked because it was the first extension that passed the `FileMatchesAssemblyName` test)